### ISSUE TITLE
A question about the evaluation

### ISSUE THREAD

**Issue by speedcell4** [yahshibu/nested-ner-tacl2020-flair#2]
Hi~

I have a question about the evaluation.
You have mentioned you ran experiments with different random seed 5 times, and you reported the averaged number on each table. But how do you calculate the averaged F1? I guess there are two methods,
1. average precision and recall scores of each experiment first, and then use these two numbers to calculate the F1 score.
2. calculate the F1 score of each experiment by using its precision and recall score first and then average these F1 scores.

But I think these two methods will result in different F1 score, which one is the method that you used in your experiments?

```python
def f1(p, r):
    return 2 * p * r / (p + r)


def average(xs):
    return sum(xs) / len(xs)


precisions = [0.1, 0.2, 0.3]
recalls = [0.4, 0.5, 0.6]

print(f1(average(precisions), average(recalls)))  # method 1

f1s = [f1(p, r) for p, r in zip(precisions, recalls)]  # method 2
print(average(f1s))

# 0.28571428571428575
# 0.28190476190476194
```

**Comment by yahshibu** [comment]
Hello. We used method 2. In addition, we calculated a standard deviation using F1 scores and their average calculated with method 2.

In this connection, we have two methods for averaging precision and recall scores.
1. sum TP, FP, and FN of each experiment first, and then use these three numbers to calculate the precision and recall scores. [micro average]
2. calculate the precision and recall scores of each experiment by using its TP, FP, and FN first, and then average these precision and recall scores. [macro average]
We adopted macro average (method 2).

**Comment by yahshibu** [comment]
Hello. We used method 2. In addition, we calculated a standard deviation using F1 scores and their average calculated with method 2.

In this connection, we have two methods for averaging precision and recall scores.

1. sum TP, FP, and FN of each experiment first, and then use these three numbers to calculate the precision and recall scores. [micro average]
2. calculate the precision and recall scores of each experiment by using its TP, FP, and FN first, and then average these precision and recall scores. [macro average]

We adopted macro average (method 2).
